backend/src/main/java/org/http/backend/controller/ScheduleController.java

```diff
@@ -37,6 +39,16 @@ public List<Integer> getBookedSeatsForSchedule(@PathVariable String scheduleId)
         return bookingService.getBookedSeatsByScheduleId(scheduleId);
     }
 
+    @GetMapping("/movie/{movieId}")
+    public ResponseEntity<List<Schedule>> findByMovieId(@PathVariable("movieId") String movieId) {
+        try {
+            List<Schedule> schedules = scheduleService.findByMovieId(movieId);
+            return ResponseEntity.ok(schedules != null ? schedules : new ArrayList<>());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
```

Comment:
We should prevent returning null values. We could return an empty list along with the error status.
```suggestion
        } catch (Exception e) {
             return ResponseEntity.internalServerError().body(Collections.emptyList());
```